python/tank/authentication/user_impl.py

```diff
@@ -206,7 +206,19 @@ def __init__(self, host, login, session_token, http_proxy, password=None, cookie
         self._login = login
         self._session_token = session_token
         self._cookies = cookies
-        self._saml_expiration = saml_expiration
+
+        # saml expiration is meaningful only with SSO enabled.
+        # The presence of cookies implies potential SSO usage.
+        if self._cookies is None:
+            self._saml_expiration = None
+        elif saml_expiration is None:
+            # For lack of accurate information, we will indicates that the
+            # credentials are expired  by setting the expiration to a moment
+            # in the past.
+            # @TODO: obtain the expiration from the cookies themselves, if possible.
```

Comment:
As for the saml_flag... We would need to open up the cookie jar (which I wanted to avoid at this time). So using the presence/absence of cookies seemed the best interim solution. At the moment, cookies are only present when we are using SSO, and the reverse logic is true.

If we were using a QWebView for all login flows (with or without SSO, thus removing the need for the login dialog), then things would be different...